crates/redis/src/lib.rs

```diff
@@ -52,30 +55,81 @@ impl TriggerExecutor for RedisTrigger {
     type RunConfig = NoArgs;
 
     async fn new(engine: TriggerAppEngine<Self>) -> Result<Self> {
-        let address = engine
+        let default_address: String = engine
             .trigger_metadata::<TriggerMetadata>()?
             .unwrap_or_default()
             .address;
 
+        let mut addresses = vec![default_address.clone()];
+
         let mut channel_components: HashMap<String, Vec<String>> = HashMap::new();
 
         for (_, config) in engine.trigger_configs() {
+            let address = config.address.clone().unwrap_or(default_address.clone());
```

Comment:
I am not sure I follow this as the address will either be the optional override or the default provided at the application level. I apologize if I am missing something obvious here.

Reply:
This comment referred to the original code, which kept cloning the default address in order to check it against the vector _which was already initialised with the default address_.  There was no need to take any action unless there was an override, because the default address was already in the vector.  Hence the suggestion to `if let Some(address) =` and do the check-and-insert only if there was an override.

The current code does depend on inserting the default address during the loop, so the comment no longer applies.

Reply:
Ah, that makes sense!